rfd/0086-stream-audit-events.md

```diff
@@ -0,0 +1,227 @@
+}
+```
+
+Next we give some information about DynamoDB streams, explaining why we have chosen such a representation for the stream cursor.
+From the DynamoDB documentation:
+
+> A stream consists of stream records.
+> Each stream record is assigned a sequence number, reflecting the order in which the record was published to the stream.
+> Stream records are organized into groups, or shards. 
+> Shards are ephemeral: They are created and deleted automatically, as needed.
+> Any shard can also split into multiple new shards; this also occurs automatically. (It's also possible for a parent shard to have just one child shard.)
+> A shard might split in response to high levels of write activity on its parent table, so that applications can process records from multiple shards in parallel.
+> Because shards have a lineage (parent and children), an application must always process a parent shard before it processes a child shard. This helps ensure that the stream records are also processed in the correct order.
+
+When starting streaming, we can use [`DescribeStream`] to retrieve a list of active stream shards:
+
+```json
+"Shards": [
+	{
+		"ParentShardId": "string",
+		"SequenceNumberRange": {
+			"EndingSequenceNumber": "string",
+			"StartingSequenceNumber": "string"
+		},
+		"ShardId": "string"
+	}
+],
+```
+
+> If the `SequenceNumberRange` has a `StartingSequenceNumber` but no `EndingSequenceNumber`, then the shard is still open (able to receive more stream records).
+> If both `StartingSequenceNumber` and `EndingSequenceNumber` are present, then that shard is closed and can no longer receive more data.
+
+For each of these shards, we also retrieve a shard iterator  using [`GetShardIterator`], providing the following information:
+
+```json
+{
+   "SequenceNumber": "string",
+   "ShardId": "string",
+   "ShardIteratorType": "string",
+   "StreamArn": "string"
+}
+```
+
+We have the following `ShardIteratorType`s:
+
+> - `AT_SEQUENCE_NUMBER` - Start reading exactly from the position denoted by a specific sequence number.
+> - `AFTER_SEQUENCE_NUMBER` - Start reading right after the position denoted by a specific sequence number.
+> - `TRIM_HORIZON` - Start reading at the last (untrimmed) stream record, which is the oldest record in the shard. In DynamoDB Streams, there is a 24 hour limit on data retention. Stream records whose age exceeds this limit are subject to removal (trimming) from the stream.
+> - `LATEST` - Start reading just after the most recent stream record in the shard, so that you always read the most recent data in the shard.
+
+Given a shard id `$ID` (returned by `DescribeStream`), if `streamCursor.ShardIdToSequenceNumber` contains `$ID`, then we set the `ShardIteratorType` to `AFTER_SEQUENCE_NUMBER` and `SequenceNumber` to `streamCursor.ShardIdToSequenceNumber[$ID]`.
+Otherwise, we can either set it to `TRIM_HORIZON` or to `LATEST`.
+In this case, we will set it to `LATEST` to maintain the current behaviour in `shards.go`.
```

Comment:
Does it mean that by settings LATEST some audit events will me omitted  and never streamed  ?